stablehlo/tests/ops_stablehlo.mlir

```diff
@@ -1420,34 +1420,18 @@ func.func @cholesky(%arg0: tensor<1x2x2xf32>) -> tensor<1x2x2xf32> {
 
 // -----
 
-func.func @cholesky_error_nonsquare(%arg0: tensor<1x2x1xf32>) -> tensor<1x2x1xf32> {
-  // expected-error@+1 {{minor dimensions of 'a' must have equal size, got shape 1, 2, 1}}
-  %0 = "stablehlo.cholesky"(%arg0) { lower = true } : (tensor<1x2x1xf32>) -> tensor<1x2x1xf32>
-  func.return %0: tensor<1x2x1xf32>
-}
-
-// -----
-
-func.func @cholesky_invalid_rank(%arg0: tensor<1xf32>) -> tensor<1xf32> {
+func.func @cholesky_c2(%arg0: tensor<1xf32>) -> tensor<1xf32> {
   // expected-error@+1 {{argument 'a' must have rank >= 2, got shape 1}}
   %0 = "stablehlo.cholesky"(%arg0) { lower = true } : (tensor<1xf32>) -> tensor<1xf32>
   func.return %0: tensor<1xf32>
 }
 
 // -----
 
-func.func @cholesky_invalid_elt(%arg0: tensor<1x2x2xi32>) -> tensor<1x2x2xi32> {
-  // expected-error@+1 {{operand #0 must be tensor of f8E4M3FN type or f8E5M2 type or f8E4M3FNUZ type or f8E5M2FNUZ type or 16-bit float or 32-bit float or 64-bit float or bfloat16 type or complex type with 32-bit float or 64-bit float elements values, but got 'tensor<1x2x2xi32>}}
-  %0 = "stablehlo.cholesky"(%arg0) { lower = true } : (tensor<1x2x2xi32>) -> tensor<1x2x2xi32>
-  func.return %0: tensor<1x2x2xi32>
-}
-
-// -----
-
-func.func @cholesky_wrong_infer_shape(%arg0: tensor<1x2x2xf32>) -> tensor<1x2x2x2xf32> {
```

Comment:
Yes we do. As per the [checklist](https://github.com/openxla/stablehlo/blob/main/docs/reference_checklist.md) no. 5, we only require one positive test, which [infer_stablehlo.mlir](https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/infer_stablehlo.mlir#L84-L90) already has.

Reply:
maybe I dd not get this. Shouldn't we need a negative test other than the positive tests mentioned below
```
5(i)
Make sure that there is at least one test for each constraint in the verifier and type inference methods; constraints covered in ODS will not be tested. These tests will mostly be negative tests exercising the fact that the constraints are not met. Make sure to add at least one positive test in case there are none.
```

Reply:
```
Make sure that there is at least one test for each constraint in the verifier
and type inference methods
```

The one test mentioned aboce would be a positive test which

```
Make sure to add at least one positive test in case there are none.
```

this statement says to add if there are none. Since we have one positive test, we've satisfied that there is at least one test for type inference.

```
These tests will mostly be negative tests exercising the fact that the constraints are not met. 
```

This just states that in most cases the tests would be negative (constraint tests) and we could have positive (type inference) tests.

Reply:
I think this requires rewording as it is not obvious and unnecessarily tricky. Will add a PR to address this.